anonlink/blocking.py

```diff
@@ -21,21 +23,22 @@ def _evalf(__funcs : 'Iterable[Callable[..., _T]]',  # https://github.com/python
     return (f(*args, **kwargs) for f in __funcs)
 
 
-class _AssociativeBinaryOp(metaclass=ABCMeta):
-    __slots__ = '_funcs'
+_Record = TypeVar('_Record')
```

Comment:
Why `BlockingFunction` but `_Record`? Shouldn't these be consistent? Shouldn't you import the `_Record` defined in `typechecking.py`?

Also I'd only put the TypeVar directly above a class declaration if it isn't used anywhere else - in this case `_Record` is used throughout the following classes so give it some whitespace to separate it from the first class.

Reply:
`_Record` is only there to parametrise the generic. It doesn’t make sense for other modules to import it. Similarly, `_Record` in typechecking.py is just there to parametrise generics. I don’t think it makes sense to import a parametrisation variable from elsewhere. But I’m open to discussion on this.

I’ve fixed the whitespace!